warehouse/router/state_generate_load_files.go

```diff
@@ -0,0 +1,79 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/rudderlabs/rudder-server/warehouse/logfield"
+
+	"github.com/rudderlabs/rudder-server/warehouse/internal/repo"
+	whutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+func (job *UploadJob) generateLoadFiles(hasSchemaChanged bool) error {
+	generateAll := hasSchemaChanged || slices.Contains(warehousesToAlwaysRegenerateAllLoadFilesOnResume, job.warehouse.Type) || job.config.alwaysRegenerateAllLoadFiles
```

Comment:
[optional] Can we break this into multiple lines?

Reply:
Is it okay, if I take this up in a follow-up PR after this PR gets merged? I wanted this standalone PR to just split the states, and later we can work on individual states.